src/extras/render-passes/render-pass-coc.js

```diff
@@ -19,37 +23,23 @@ class RenderPassCoC extends RenderPassShaderQuad {
         super(device);
         this.cameraComponent = cameraComponent;
 
-        const screenDepth = ChunkUtils.getScreenDepthChunk(device, cameraComponent.shaderParams);
-        this.shader = this.createQuadShader(`CocShader-${nearBlur}`, /* glsl */`
+        // register shader chunks
+        ShaderChunks.get(device, SHADERLANGUAGE_GLSL).set('CocPS', glslCocPS);
+        ShaderChunks.get(device, SHADERLANGUAGE_WGSL).set('CocPS', wgslCocPS);
 
-            ${nearBlur ? '#define NEAR_BLUR' : ''}
-            ${screenDepth}
-            varying vec2 uv0;
-            uniform vec3 params;
+        const defines = new Map();
+        if (nearBlur) defines.set('NEAR_BLUR', '');
 
-            void main()
-            {
-                float depth = getLinearScreenDepth(uv0);
+        // add defines needed for correct use of screenDepthPS chunk
+        ShaderUtils.addScreenDepthChunkDefines(device, cameraComponent.shaderParams, defines);
 
-                // near and far focus ranges
-                float focusDistance = params.x;
-                float focusRange = params.y;
-                float invRange = params.z;
-                float farRange = focusDistance + focusRange * 0.5;
-                
-                // near and far CoC
-                float cocFar = min((depth - farRange) * invRange, 1.0);
-
-                #ifdef NEAR_BLUR
-                    float nearRange = focusDistance - focusRange * 0.5;
-                    float cocNear = min((nearRange - depth) * invRange, 1.0);
-                #else
-                    float cocNear = 0.0;
-                #endif
-
-                gl_FragColor = vec4(cocFar, cocNear, 0.0, 0.0);
-            }`
-        );
+        this.shader = ShaderUtils.createShader(device, {
+            uniqueName: `CocShader-${nearBlur}`,
+            attributes: { aPosition: SEMANTIC_POSITION },
+            vertexChunk: 'RenderPassQuadVS',
+            fragmentChunk: 'CocPS',
```

Comment:
Shouldn't this be `cocPS`?

Reply:
good point, all chunk names start with a lower case character. Let me fix, there's about 5 of these.